block/src/Pos/Block/Behavior.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE RankNTypes #-}
+module Pos.Block.Behavior
+       ( BlockBehavior (..)
+       , ForgeHeaderParams (..)
+       , HasBlockBehavior
+       , withBlockBehavior
+       , blockBehavior
+       ) where
+
+import           Universum
+
+import qualified Data.Aeson as A
+import           Data.Default (Default (..))
+import           Data.Reflection (Given (..), give, given)
+import           Serokell.Aeson.Options (defaultOptions)
+
+import           Pos.Util.Util (toAesonError)
+
+----------------------------------------------------------------------------
+-- Types for the behavior config
+----------------------------------------------------------------------------
+
+-- | Block settings (a part of the behavior config).
+--
+-- The syntax of this config section is as follows:
+--
+-- @
+-- block:
+--     forgeHeader: Normal | WrongLeader
+-- @
+data BlockBehavior = BlockBehavior
+    { -- | Block header forging settings
+      bbForgeHeader :: !ForgeHeaderParams
+    }
+    deriving (Eq, Show, Generic)
+
+data ForgeHeaderParams
+    = HeaderNormal      -- ^ Do not forge anything
+    | HeaderWrongLeader -- ^ Create blocks header with a wrong slot leader
+                        --    signature
+    deriving (Eq, Show)
+
+type HasBlockBehavior = Given BlockBehavior
```

Comment:
Hm, why did you decide to use `reflection` instead of passing it in `ReaderT` context? :/
We mostly use `ReaderT`. For example, `BehaviorConfig` is currently stored in `ReaderT` context.
AFAIU, there are only two good reasons to use `reflection`:
1. If there is a lot of existing code which uses some constant `x` and you want to make `x` configurable.
2. If you use `x` in some instance (e. g. `Bi`) where you want to know value of this `x`, but can't use any external information and `x` is not constant.

Reply:
So, AFAIU, now there are two ways to access `BlockBehavior`: take it from reader context or take it using `reflection`. And it's not guaranteed that they will be the same (one can use `local` to modify one of them, but the one in `reflection` can not be modified). Sounds like a bad design.

Reply:
In #2562 I added `UpdateBehavior` which is stored only in reader and it was quite easy.